Bottom: 78eb63a4fa09ed466e93642794e9423c4278f523
Top:    8a0bc35ec2704699ed4b1466444ed762e1a37f78
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 14:20:54 +0800

autofs-5.0.7 - fix leaked ldap percent hack allocation in lookup_one()

Fix a resource leak when calling the percent hack transforms in lookup_one().


---

diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 3371b3f..f8feed5 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -2525,6 +2525,7 @@ static int lookup_one(struct autofs_point *ap,
 		if (enc_len1 != 0) {
 			enc_len2 = encode_percent_hack(qKey, &enc_key2, 1);
 			if (enc_len2 < 0) {
+				free(enc_key1);
 				crit(ap->logopt,
 				     "could not use percent hack encode key %s",
 				     qKey);
